Tangem/Modules/ManageTokens/ManageTokensView.swift

```diff
@@ -9,17 +9,78 @@
 import SwiftUI
 import Combine
 import BlockchainSdk
+import AlertToast
 
 struct ManageTokensView: View {
     @ObservedObject var viewModel: ManageTokensViewModel
 
     var body: some View {
-        Text(" ManageTokensView - InProgress")
+        ZStack {
+            list
+
+            overlay
+        }
+        .scrollDismissesKeyboardCompat(true)
+        .navigationBarTitle(Text(Localization.addTokensTitle), displayMode: .automatic)
+        .alert(item: $viewModel.alert, content: { $0.alert })
+        .toast(isPresenting: $viewModel.showToast) {
+            AlertToast(type: .complete(Colors.Icon.accent), title: Localization.contractAddressCopiedMessage)
+        }
+        .searchableCompat(text: $viewModel.enteredSearchText.value)
+        .background(Colors.Background.primary.edgesIgnoringSafeArea(.all))
+        .onAppear { viewModel.onAppear() }
+        .onDisappear { viewModel.onDisappear() }
     }
-}
 
-struct ManageTokensView_Previews: PreviewProvider {
-    static var previews: some View {
-        ManageTokensView(viewModel: ManageTokensViewModel())
+    private var list: some View {
+        ScrollView {
+            LazyVStack(spacing: 0) {
+                if #available(iOS 15.0, *) {} else {
+                    SearchBar(text: $viewModel.enteredSearchText.value, placeholder: Localization.commonSearch)
+                        .padding(.horizontal, 8)
+                        .listRowInsets(.init(top: 8, leading: 16, bottom: 8, trailing: 16))
+                }
+
+                ForEach(viewModel.tokenViewModels) {
+                    ManageTokensItemView(viewModel: $0)
+                }
+
+                if viewModel.hasNextPage {
+                    HStack(alignment: .center) {
+                        ActivityIndicatorView(color: .gray)
+                            .onAppear(perform: viewModel.fetch)
+                    }
+                }
+
+                Color.clear.frame(width: 10, height: 58, alignment: .center)
```

Comment:
```suggestion
                Color.clear.frame(height: 58)
```
выглядит что так должно сработать